Removed HPA objects from extensions api group

piosz opened the pull request:
fix #29778

``` release-note
HorizontalPodAutoscaler is no longer supported in extensions/v1beta1 version. Use autoscaling/v1 instead.
```

 cc @kubernetes/autoscaling

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.kubernetes.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/35782)

<!-- Reviewable:end -->

piosz commented:
We need to remove client 1.3 first. I'll create another PR

piosz committed:
Removed HPA objects from extensions api group

piosz commented:
@k8s-bot kops aws e2e test this

piosz commented:
This is now ready for the review.
@caesarxuchao @janetkuo @DirectXMan12 could you please take a look?

cc @kubernetes/sig-autoscaling-misc @kubernetes/sig-api-machinery-misc

DirectXMan12 commented:
What's up with removing `test-update-storage-objects.sh`?  Can you please put an explanation in the commit message?

DirectXMan12 commented:
otherwise, this generally seems good to me. 🎉

DirectXMan12 reviewed (changes requested):
What's up with removing `test-update-storage-objects.sh`?  Can you please put an explanation in the commit message?

Otherwise, this generally LGTM 🎉

piosz commented:
@DirectXMan12 I'm not sure whether we want to remove it but IIUC testing upgrade path of HPA was the last remaining test case there.

DirectXMan12 commented:
> @DirectXMan12 I'm not sure whether we want to remove it but IIUC testing upgrade path of HPA was the last remaining test case there.

ack, makes sense, but it'll probably be needed again for other API groups later.  At least put that explanation in the commit message.

piosz commented:
I'll rebase autogenerated files right before merge, because they are changing frequently.

@DirectXMan12 I'll update during rebase.

@lavalamp @kubernetes/sig-api-machinery-misc it seems @caesarxuchao is unavailable, can one of you take a look? 
@janetkuo could you please take a look into kubectl changes?

janetkuo reviewed (commented):
kubectl changes LGTM

janetkuo commented:
I looked at bootstrappolicy and it's still referencing hpa under extensions api group. Maybe you want to change that too?

piosz committed:
Removed hpa v1beta1 from bootstrappolicy

piosz commented:
@DirectXMan12 added a comment about test-upgrade-storage into commit message. Thanks!
@janetkuo thanks for review! I've addresses your comment about bootstrappolicy.

@lavalamp @kubernetes/sig-api-machinery-misc rebasing this PR all the time is a huge pain. Could you please take a look?

lavalamp commented on hack/test-update-storage-objects.sh:
```diff
@@ -1,223 +0,0 @@
-# See the License for the specific language governing permissions and
-# limitations under the License.
-
-# Script to test cluster/update-storage-objects.sh works as expected.
```
Do not delete this file. It is testing the etcd2-3 migration in addition to the super important file that touches everything to rewrite it in a new version after an update.

piosz replied:
ACK

lavalamp commented on hack/jenkins/test-dockerized.sh:
```diff
@@ -57,4 +57,3 @@ go install ./cmd/...
 
 make test-cmd
 make test-integration
-./hack/test-update-storage-objects.sh
```
Do not stop running this test

piosz replied:
ACK

lavalamp commented:
I'm completely swamped, I don't even have time to find someone to review, sorry. Please don't block on me.

...but also don't delete that test file, that makes me a bit worried that maybe I should look at the rest of this PR. Which I definitely don't have time to do.

piosz committed:
Autogenerated changes

piosz commented:
Rebased and reverted changes which removes test-update-storage-objects. Reassigning to @wojtek-t

janetkuo commented on plugin/pkg/auth/authorizer/rbac/bootstrappolicy/policy.go:
```diff
@@ -145,7 +145,9 @@ func ClusterRoles() []rbac.ClusterRole {
 				rbac.NewRule(ReadWrite...).Groups(extensionsGroup).Resources("daemonsets", "deployments", "deployments/scale",
-					"horizontalpodautoscalers", "ingresses", "replicasets", "replicasets/scale", "replicationcontrollers/scale").RuleOrDie(),
+					"ingresses", "replicasets", "replicasets/scale", "replicationcontrollers/scale").RuleOrDie(),
+				rbac.NewRule(ReadWrite...).Groups(extensionsGroup).Resources("daemonsets", "replicationcontrollers/scale",
```
Why repeat this?

piosz replied:
Ups, probably a side effect of one of thousand rebases here. I'll fix this. Thanks for catching!

wojtek-t commented on hack/test-update-storage-objects.sh:
```diff
@@ -107,16 +107,17 @@ echo "${ETCD_VERSION}/${STORAGE_BACKEND_ETCD2}" > "${ETCD_DIR}/version.txt"
 ### BEGIN TEST DEFINITION CUSTOMIZATION ###
 
+# After removing HPA in v1beta1 there is basically no resource which has
+# two versions. In the future we may want to extend it.
```
@lavalamp - I think this is the right thing to do. WDYT?

caesarxuchao replied:
Job has v1 and v2alpha1, can we test that? I feel it's unfair to ask you to test Job, but otherwise we don't test any version upgrades.

@soltysh it looks like you are the last one that touched this file, do you remember if we should test v1 and v2alpha1 Job in this script?

piosz replied:
Created #40860

cc @DirectXMan12 wh

soltysh replied:
Jobs should be removed from v2alpha1 per https://github.com/kubernetes/kubernetes/issues/37166

wojtek-t commented on pkg/kubectl/describe.go:
```diff
@@ -136,7 +136,6 @@ func describerMap(c clientset.Interface) map[schema.GroupKind]Describer {
 		api.Kind("ConfigMap"):             &ConfigMapDescriber{c},
 
 		extensions.Kind("ReplicaSet"):                  &ReplicaSetDescriber{c},
-		extensions.Kind("HorizontalPodAutoscaler"):     &HorizontalPodAutoscalerDescriber{c},
```
Don't we want to have this one?

piosz replied:
Yes, here is another describer `autoscaling.Kind("HorizontalPodAutoscaler")`

piosz commented:
Rebased

caesarxuchao commented:
One question about the test file, otherwise lgtm.

piosz commented:
@caesarxuchao thanks for the review. I'll rebase soon.

wojtek-t commented:
/lgtm
/approve

piosz commented:
@k8s-bot cri e2e test this
